### PR TITLE
BACKLOG-13857 : add onChange on text selector

### PR DESCRIPTION
**Pull request by dgriffon**
https://jira.jahia.org/browse/BACKLOG-13857

### CODE REVIEW

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Text/Text.jsx [hunk 1]
```diff
@@ -4,16 +4,17 @@ import {Input} from '@jahia/design-system-kit';
...
+export const TextCmp = ({field, value, id, editorContext, onChange}) => {
     const fieldType = field.requiredType;
     const isNumber = fieldType === 'DOUBLE' || fieldType === 'LONG' || fieldType === 'DECIMAL';
     const decimalSeparator = editorContext.uilang === 'en' ? '.' : ',';
-    const controlledValue = value === undefined ? '' : value;
+    const controlledValue = value === undefined ? '' : (isNumber ? value && value.replace('.', decimalSeparator) : value);
     return (
         <FastField render={({form: {handleChange, setFieldTouched}}) => {
-            const onChange = evt => {
+            const onFieldChangeChange = evt => {
```
Comment (Taybou) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Text/Text.jsx:14]:
```suggestion
            const onFieldChange = evt => {
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Text/Text.jsx [hunk 2]
```diff
@@ -4,16 +4,17 @@ import {Input} from '@jahia/design-system-kit';
 import {FieldPropTypes} from '~/FormDefinitions/FormData.proptypes';
 import {FastField} from 'formik';
 
-export const TextCmp = ({field, value, id, editorContext}) => {
+export const TextCmp = ({field, value, id, editorContext, onChange}) => {
     const fieldType = field.requiredType;
     const isNumber = fieldType === 'DOUBLE' || fieldType === 'LONG' || fieldType === 'DECIMAL';
     const decimalSeparator = editorContext.uilang === 'en' ? '.' : ',';
-    const controlledValue = value === undefined ? '' : value;
+    const controlledValue = value === undefined ? '' : (isNumber ? value && value.replace('.', decimalSeparator) : value);
```
Comment (Taybou) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Text/Text.jsx:11]:
```suggestion
    const controlledValue = value === undefined ? '' : (isNumber ? value?.replace('.', decimalSeparator) : value);
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Text/Text.jsx [hunk 3]
```diff
@@ -25,12 +26,12 @@ export const TextCmp = ({field, value, id, editorContext}) => {
...
                         'aria-required': field.mandatory
                     }}
-                    value={isNumber ? controlledValue && controlledValue.replace('.', decimalSeparator) : controlledValue}
+                    value={controlledValue}
                     readOnly={field.readOnly}
                     type={isNumber ? 'number' : 'text'}
                     decimalScale={fieldType === 'LONG' ? 0 : undefined}
                     decimalSeparator={decimalSeparator}
-                    onChange={onChange}
+                    onChange={onFieldChangeChange}
```
Comment (Taybou) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Text/Text.jsx:34]:
```suggestion
                    onChange={onFieldChange}
```

File: src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Text/Text.test.jsx [hunk 4]
```diff
@@ -14,6 +14,7 @@ describe('Text component', () => {
             editorContext: {
                 uilang: 'en'
             },
+            onChange: () => {},
```
Comment (Taybou) [inline src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Text/Text.test.jsx:17]: you can mock the onChange function with jest:fn() and validate if it's called or not (ex: https://github.com/Jahia/content-editor/pull/706/files#diff-90d9f6486bd3030f7dd2d7efee4feed9R92)

Comment (dgriffon) [reply src/javascript/EditPanel/EditPanelContent/FormBuilder/Section/FieldSet/Field/SelectorTypes/Text/Text.test.jsx]: done, thank you for the hint!

**Pull request merged**